darwin/dataset/utils.py

```diff
@@ -0,0 +1,291 @@
+                          stratify=labels)
+    )
+    X_val, X_test, y_val, y_test = remove_cross_contamination(
+        *train_test_split(X_tmp, y_tmp,
+                          test_size=test_percentage / (val_percentage + test_percentage),
+                          random_state=split_seed,
+                          stratify=y_tmp)
+    )
+    # Remove duplicates within the same set
+    # NOTE: doing that earlier (e.g. in remove_cross_contamination()) would produce mathematical
+    # mistakes in the class balancing between validation and test sets.
+    return list(set(X_train)), list(set(X_val)), list(set(X_test))
+
+def split_dataset(
+        dataset,
+        val_percentage: Optional[float] = 0.1,
+        test_percentage: Optional[float] = 0.2,
+        force_resplit: Optional[bool] = False,
+        split_seed: Optional[int] = 42,
+):
+    """
+    Given a local a dataset (pulled from Darwin) creates lists of file names
+    for each split for train, validation, and test.
+
+    Parameters
+    ----------
+    dataset : TODO
+        Dataset to split
+    val_percentage : float
+        Percentage of images used in the validation set
+    test_percentage : float
+        Percentage of images used in the test set
+    force_resplit : bool
+        Discard previous split and create a new one
+    split_seed : in
+        Fix seed for random split creation
+
+    Returns
+    -------
+    splits : dict
+        Keys are the different splits (random, tags, ...) and values are the relative file names
+    """
+    assert dataset is not None
+    annotation_path = Path(dataset.local_path / "annotations")
+    assert annotation_path.exists()
+    annotation_files = list(annotation_path.glob("*.json"))
+
+    # Extract list of classes and create respective files
+    lists_path = dataset.local_path / "lists"
+    lists_path.mkdir(parents=True, exist_ok=True)
+    idx_to_classes_polygon = make_class_list(
+        "classes_polygon.txt", annotation_files, lists_path, "polygon", force_resplit, add_background=True
+    )
+    idx_to_classes_tags = make_class_list(
+        "classes_tags.txt", annotation_files, lists_path, "tag", force_resplit
+    )
+
+    # Create split id, path and final split paths
+    assert val_percentage is not None
+    assert test_percentage is not None
+    assert split_seed is not None
+    assert 0 < val_percentage < 1.0
+    assert 0 <= test_percentage < 1.0
+    assert val_percentage + test_percentage < 1.0
+    split_id = f'split_v{int(val_percentage*100)}_t{int(test_percentage*100)}_s{split_seed}'
+    split_path = lists_path / split_id
+
+    splits = {}
+    # Do the actual split
+    if not split_path.exists() or force_resplit:
+        os.makedirs(str(split_path), exist_ok=True)
+
+        # RANDOM SPLIT
+        train_path = Path(split_path / "random_split_train.txt")
+        val_path = Path(split_path / "random_split_val.txt")
+        test_path = Path(split_path / "random_split_test.txt")
```

Comment:
if `split_path` is already of the type Path, why do you need to wrap it around in `Path(split_path / "....")` ?